### PR TITLE
[TEP-0093] Add sign and verify as experiment subcommands to tekton cli 

### PR DESCRIPTION
**Pull request by Yongxuanzhang**
# Changes

This commit adds sign and verify as experiment subcommands for task and pipeline. Examples:
```bash
tkn task sign examples/example-task.yaml -K=cosign.key -f=signed.yaml
tkn task verify examples/signed.yaml -K=cosign.pub
```

Signed-off-by: Yongxuan Zhang yongxuanzhang@google.com

/kind feature

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Run the code checkers with `make check`
- [x] Regenerate the manpages, docs and go formatting with `make generated`
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/cli/blob/master/CONTRIBUTING.md)
for more details._

# Release Notes

### CODE REVIEW

**Comment by vdemeester** [comment]
cc @concaf

File: docs/cmd/tkn_pipeline_sign.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+Sign a Pipeline pipeline.yaml:
+	tkn pipeline sign pipeline.yaml -K=cosign.key -f=signed.yaml
+or using kms
+	tkn pipeline sign pipeline.yaml -K=gcpkms://projects/PROJECTID/locations/LOCATION/keyRings/KEYRING/cryptoKeys/KEY/cryptoKeyVersions/VERSION -f=signed.yaml
+
+### Options
+
+```
+      --allow-missing-template-keys   If true, ignore any errors in templates when a field or map key is missing in the template. Only applies to golang and jsonpath output formats. (default true)
+  -f, --file-name string              Skip verifying the payload'signature
```
Comment (QuanZhang-William) [inline docs/cmd/tkn_pipeline_sign.md:26]: the description does not seem to match --file-name? 🤔

File: docs/cmd/tkn_pipeline_sign.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  -h, --help                          help for sign
+  -K, --key-file string               Key file
+  -m, --kms-key string                Skip verifying the payload'signature
```
Comment (QuanZhang-William) [inline docs/cmd/tkn_pipeline_sign.md:29]: same as above and a couple of same issue in the below files if it is the case

File: pkg/cmd/pipeline/sign.go [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+				return err
+			}
+
+			crd := &v1beta1.Pipeline{}
+			if err := yaml.Unmarshal(b, &crd); err != nil {
+				return fmt.Errorf("error unmarshalling Task: %v", err)
+			}
+
+			// Sign the task and save to file
+			if err := trustedresources.Sign(crd, opts.keyfile, opts.kmsKey, opts.targetFile); err != nil {
```
Comment (QuanZhang-William) [inline pkg/cmd/pipeline/sign.go:69]: Can I use the same file as input and output if I want to overwrite? Maybe worth adding this point to the user guide.

File: pkg/cmd/task/sign.go [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+			if err := trustedresources.Sign(crd, opts.keyfile, opts.kmsKey, opts.targetFile); err != nil {
+				return fmt.Errorf("error signing Task: %v", err)
+			}
+			fmt.Fprintf(s.Out, "Task %s is signed successfully \n", args[0])
+			return nil
+		},
+	}
+	f.AddFlags(c)
+	c.Flags().StringVarP(&opts.keyfile, "key-file", "K", "", "Key file")
+	c.Flags().StringVarP(&opts.kmsKey, "kms-key", "m", "", "Skip verifying the payload'signature")
```
Comment (QuanZhang-William) [inline pkg/cmd/task/sign.go:82]: Does kms-key means "Skip verifying the payload'signature? 🤔

**Review by QuanZhang-William — COMMENTED** [review]
Nice work! Looking forward to use it! 🚀

File: docs/cmd/tkn_pipeline_sign.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+### Synopsis
+
+Sign Tekton Task/Pipeline
+
+### Examples
+
+Sign a Pipeline pipeline.yaml:
+	tkn pipeline sign pipeline.yaml -K=cosign.key -f=signed.yaml
+or using kms
+	tkn pipeline sign pipeline.yaml -K=gcpkms://projects/PROJECTID/locations/LOCATION/keyRings/KEYRING/cryptoKeys/KEY/cryptoKeyVersions/VERSION -f=signed.yaml
```
[the remark this answers is not in the document]
Comment (Yongxuanzhang) [reply docs/cmd/tkn_pipeline_sign.md:20]: Oh thanks for the reminder! I will add it

File: docs/cmd/tkn_pipeline_sign.md (hunk 1 above)
Comment (Yongxuanzhang) [reply docs/cmd/tkn_pipeline_sign.md:26]: 😂  good catch

File: pkg/cmd/pipeline/sign.go (hunk 3 above)
Comment (Yongxuanzhang) [reply pkg/cmd/pipeline/sign.go:69]: sure!

File: pkg/cmd/task/sign.go (hunk 4 above)
Comment (Yongxuanzhang) [reply pkg/cmd/task/sign.go:82]: Thanks!

**Comment by vdemeester** [comment]
I wonder if we should "put" those commands under experimental (see https://github.com/tektoncd/cli/pull/1575). Otherwise, nice !
/approve

**Comment by concaf** [comment]
looks good :)

**Comment by Yongxuanzhang** [comment]
> I wonder if we should "put" those commands under experimental (see #1575). Otherwise, nice ! /approve

Yes that sounds good! I will put them under experimental!

**Comment by vdemeester** [comment]
> > I wonder if we should "put" those commands under experimental (see #1575). Otherwise, nice ! /approve
> 
> Yes that sounds good! I will put them under experimental! But should we merge that PR first?

We should probably merge #1575 first indeed 👼🏼 
cc @pradeepitm12 @piyush-garg

**Comment by bendory** [comment]
please fix typo -- "tekton", not "tektont" in the title

**Pull request merged**